bash_functions.sh

```diff
@@ -46,7 +46,7 @@ validate_env() {
     # Optional ServerIP is a valid IP
     # nc won't throw any text based errors when it times out connecting to a valid IP, otherwise it complains about the DNS name being garbage
     # if nc doesn't behave as we expect on a valid IP the routing table should be able to look it up and return a 0 retcode
-    if [[ "$(nc -4 -w1 -z "$ServerIP" 53 2>&1)" != "" ]] || ! ip route get "$ServerIP" > /dev/null ; then
+    if [[ "$(nc -4 -w1 -z "$ServerIP" 53 2>&1)" != "" ]] && ! ip route get "$ServerIP" > /dev/null ; then
```

Comment:
> if nc doesn't behave as we expect on a valid IP the routing table should be able to look it up and return a 0 retcode

The comment states netcat can be unreliable so ip route is a fallback, so it should be ||, not && shouldn't it?

Reply:
no, if `nc` fails with an `||`, then `ip route get` will never run. `&&` means that both will run each time, and if both `nc` and `ip route get` fail, then we know for sure it is no good. If either of them pass, then we can be reasonably sure its valid.

Reply:
The better solution would be to figure out why `armhf:buster` & `arm64:buster` has an issue with `ip route get`. I spent some time on it, but when I starting running into things involving `qemu`, I gave up.

Reply:
The confusing part is the they are both negative tests, so `!=` && `!=`